src/collective/privacy/browser/consent.py

```diff
@@ -117,6 +114,6 @@ def getConsentRequired(self):
             try:
                 if not reason.isOpinionExpressed(self.request):
                     found.append(reason)
-            except:
+            except Exception:
```

Comment:
this is not a good, but I have no idea on what's the expected exception… can you leave a `# FIXME` comment?

Reply:
code-analysis was complaining about the bare exception but I didn't even know which exceptions were expected...

Reply:
The only exception that is currently being raised is `ValueError`, which happens when trying to get the opinion or consent state of a user that isn't possible with the underlying configuration.

For example, if the item is stored in a cookie then it's only possible to find with `user=None`, if it's stored keyed on email then it's not always possible to determine with `user=None`. I'm not sure if `ValueError` is the right semantic choice here, the other potential is `KeyError` but I think that's a bit too far from the actual semantics of the call.